site-src/geps/gep-917.md

```diff
@@ -126,23 +124,46 @@ Fields marked "Extended" support will eventually have conformance tests that
 lock in the behavior of that feature, and there will be a mechanism for implementations
 to tell the testing framework what extended fields they support.
 
-#### Testing framework
+## Testing framework
+
+Conformance tests will initially be structured as a CLI that runs on existing
+Kubernetes clusters. This may be expanded to include a testing library that can
+be imported by implementations. This tool will output test results in a
+structured format that can be used to generate reports. Where possible it will
+share code with the existing ingress-controller-conformance test suite.
+
+### Test Definitions
+
+Each set of tests will be built around a set of YAML manifests representing
+Gateway API resources. The tests will focus on how they expect a given
+controller to process those resources, configured via GatewayClass name. For
+example, tests may check to ensure that a controller correctly populates status,
+or that a request is appropriately routed to the desired backend with any
+relevant modifications.
```

Comment:
This vaguely reminds me of https://github.com/kudobuilder/kuttl, which I worked on with some friends at a previous job. While I don't necessarily suggest we use `kuttl` itself it might be worth reviewing it for ideas and influences if nothing else.

Reply:
Thanks! I took a look at kuttl and it has some pretty great patterns. I think our conformance tests will need to be somewhat more complex, specifically making potentially advanced requests and ensuring their responses match what we'd expect. This does seem like a great reference point for any of our tests that will start with a set of Kubernetes resources and expect status to be populated in a specific way.